Applications/ClientWebPortal/Source/webapp/src/utils/table-models/portfolio-model/index.js

```diff
@@ -0,0 +1,130 @@
+import Model from 'utils/table-models/model';
+import IndexedModel from 'utils/table-models/indexed-model';
+
+const KEY_INDICES = [0, 1, 10];
+const COLUMN_NAMES = [
+  'Account',
+  'Security',
+  'Quantity',
+  'Side',
+  'Average Price',
+  'Total P&L',
+  'Unrealized P&L',
+  'Realized P&L',
+  'Fees',
+  'Cost Basis',
+  'Currency',
+  'Volume',
+  'Trades',
+  'Acc. Total P&L',
+  'Acc. Unrealized P&L',
+  'Acc. Fees'
+];
+
+export default class extends Model {
+  constructor(riskServiceClient) {
+    super();
+    this.riskServiceClient = riskServiceClient;
+    this.indexedModel = new IndexedModel(KEY_INDICES, COLUMN_NAMES);
+    this.onDataReceived = this.onDataReceived.bind(this);
+
+    this.riskServiceClient.subscribePortfolio(this.onDataReceived)
+      .then(subscriptionId => {
+        this.subscriptionId = subscriptionId;
+      });
+  }
+
+  getRowCount() {
+    return this.indexedModel.getRowCount();
+  }
+
+  getColumnCount(columnIndex) {
+    return this.indexedModel.getColumnCount(columnIndex);
+  }
+
+  getValueAt(x, y) {
+    return this.indexedModel.getValueAt(x, y);
+  }
+
+  getColumnName(columnIndex) {
+    return this.indexedModel.getColumnName(columnIndex);
+  }
+
+  addDataChangeListener(listener) {
+    return this.indexedModel.addDataChangeListener(listener);
+  }
+
+  removeDataChangeListener(subId) {
+    this.indexedModel.removeDataChangeListener(subId);
+  }
+
+  dispose() {
+    this.riskServiceClient.unsubscribePortfolio(this.subscriptionId);
+  }
+
+  /** @private */
+  onDataReceived(data) {
+    for (let i=0; i<data.length; i++) {
+      let rowData = Object.freeze(this.toRowData(data[i]));
```

Comment:
This freeze doesn't do anything.

Reply:
I was actually contemplating on this one. I figured since I was passing an array to downstream, I prevent it from it being modified since this is supposed to be just "data snapshot".

Reply:
But this class doesn't use rowData for any purpose other than passing it into the IndexedModel.  Once it gives it to the IndexedModel it should be none of its business what IndexedModel does behind the scenes so long as IndexedModel honors its contract.

Reply:
I see what you are saying. I guess it was more like in case it is used anywhere else but that's a concern for the future. I will fix this up.